lib/types.nix

```diff
@@ -113,4 +113,32 @@ rec {
         types.optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType
       }";
     };
+
+  pluginLuaConfig = types.submodule (
+    { config, ... }:
+    {
+      options = {
+        pre = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration before the initialization of the plugin";
```

Comment:
I think this description could clarify that `pre` is actually merged into `init` with a "before" merge-order priority

Reply:
Seems like an implementation level detail that would cloud the purpose of the option.

Reply:
I don't think the fact that `pre` and `post` are merged into `init` is purely an implementation detail:

- End users will be exposed to this detail when _reading_ the final `init` value
- `mkOrder` is a user-facing utility
  - how mkOrder interacts with pre & post could be important to some users

I think the implementation is fine, so long as users aren't surprised by it.

Reply:
>     * End users will be exposed to this detail when _reading_ the final `init` value

True, this makes sense to me as a reason to include some mention, then.